src/components/ebay-combobox/index.js

```diff
@@ -0,0 +1,179 @@
+        });
+    },
+    init() {
+        observer.observeRoot(this, ['disabled'], () => {
+            this.expander.expandOnClick = !this.state.disabled;
+        });
+
+        this.getEls('option').forEach((optionEl, i) => {
+            Object.defineProperty(optionEl, 'selected', {
+                get: () => this.state.selectedIndex === i,
+                set: (value) => this.setSelectedIndex(value ? i : null)
+            });
+        });
+    },
+    onRender() {
+        const comboboxInput = this.getEl('input');
+        const optionsContainer = this.getEl('options');
+        const selectedIndex = findIndex(this.state.options, option => option.selected);
+
+        if (this.state.options.length) {
+            this.activeDescendant = ActiveDescendant.createLinear(
+                this.el,
+                comboboxInput,
+                optionsContainer,
+                '.combobox__option[role=option]', {
+                    activeDescendantClassName: 'combobox__option--active',
+                    autoInit: selectedIndex === -1 ? -1 : 0,
+                    autoReset: -1
+                }
+            );
+
+            this.expander = new Expander(this.el, {
+                autoCollapse: true,
+                expandOnFocus: true,
+                expandOnClick: this.state.readonly && !this.state.disabled,
+                collapseOnFocusOut: !this.state.readonly,
+                contentSelector: `#${optionsContainer.id}`,
+                hostSelector: `#${this.getEl('input').id}`,
+                expandedClass: 'combobox--expanded',
+                simulateSpacebarClick: true
+            });
+        }
+
+        this.moveCursorToEnd();
+    },
+    onBeforeUpdate() {
+        if (this.activeDescendant) {
+            this.activeDescendant.destroy();
+        }
+
+        if (this.expander) {
+            this.expander.cancelAsync();
+        }
+    },
+    onDestroy() {
+        this.activeDescendant.destroy();
+        this.expander.cancelAsync();
+    },
+    handleExpand() {
+        elementScroll.scroll(this.getEls('option')[this.state.selectedIndex]);
+        this.moveCursorToEnd();
+        emitAndFire(this, 'combobox-expand');
+    },
+    handleCollapse() {
+        this.emitChangeEvent();
+        emitAndFire(this, 'combobox-collapse');
+    },
+    moveCursorToEnd() {
+        const currentInput = this.getEl('input');
+
+        if (currentInput) {
+            const len = currentInput.value.length;
+            currentInput.setSelectionRange(len, len);
+        }
+    },
+    handleComboboxKeyUp(originalEvent) {
+        const selectedEl = this.getEl('options').querySelector('.combobox__option--active');
+        let newValue = this.getEl('input').value;
+
+        eventUtils.handleUpDownArrowsKeydown(originalEvent, () => {
+            if (!this.expander.isExpanded() && this.getEls('option').length > 0) {
+                this.expander.expand();
+            }
+            this.moveCursorToEnd();
```

Comment:
Would this cause issues where the user is trying to go back and edit the input and it would always move them to the end?

Reply:
This is only for the up and down arrows, which usually move you all the way to the beginning or all the way to the end, respectively. If we didn't have this, when a user would press the up arrow the cursor would always move to the very beginning. Maybe I have to handle the up and down arrows separately in this use case?

FWIW: Right and left arrows would still allow them to traverse the content as normal.